great_expectations/data_context/store/store_backend.py

```diff
@@ -263,6 +264,22 @@ def _has_key(self, key):
     def remove_key(self, key) -> None:
         del self._store[key]
 
+    def build_key(self, name: Optional[str], id_: Optional[str]) -> DataContextKey:
+        """
+        Utility to build a conditionally build specific key types based on input arguments.
+
+        Args:
+            name: The name of the object to be stored
+            id_: The id of the object to be stored
+
+        Returns:
+            key: The key generated as a result of the input name and id_
+
+        Raises:
+            NotImplementedError if not defined by child classes in the StoreBackend hierarchy
```

Comment:
I think this should be optional currently for other StoreBackends

Reply:
So are you saying we should remove the `raise`? I was originally going to use `pass` but I think debugging could be difficult if we don't immediately yell at the caller for using the unimplemented method.

The docstring should definitely signal that this is optional though.

Reply:
@anthonyburdi 
As long as it isn't decorated as an `abcmethod` it is "optional".

I think ` raise NotImplementedError` is appropriate here.
This just ensures that we catch cases where someone attempts to call `build_key()` but it doesn't force implementation of this method if it would never be called.

Reply:
For sure re: absence of decorator - I think this line just threw me off `NotImplementedError if not defined by child classes in the StoreBackend hierarchy`